apps/cotacoes_etl/lib/cotacoes_etl/workers/pesagro/cotacao_converter.ex

```diff
@@ -0,0 +1,154 @@
+  end
+
+  @spec trigger_txt_download(Job.t(), integer) :: :ok
+  def trigger_txt_download(job, time_offset) do
+    :timer.apply_after(second_to_ms(time_offset), GenServer, :cast, [
+      __MODULE__,
+      {:download_txt, job}
+    ])
+
+    :ok
+  end
+
+  @impl true
+  def init(_) do
+    unless File.exists?(@storage_path) do
+      File.mkdir_p!(@storage_path)
+    end
+
+    {:ok, nil}
+  end
+
+  @impl true
+  def handle_cast(:convert, _state) do
+    Logger.info(
+      "[#{__MODULE__}] ==> Iniciando a conversão pdf->txt de cotações pendentes da Pesagro"
+    )
+
+    cotacoes_to_upload = CotacaoHandler.find_cotacoes_not_downloaded()
+
+    Logger.info(
+      "#{__MODULE__} ==> #{length(cotacoes_to_upload)} cotações para serem convertidas da Pesagro"
+    )
+
+    tasks =
+      for cotacao <- cotacoes_to_upload do
+        Process.sleep(second_to_ms(1))
+
+        Task.async(fn ->
+          file_path = pesagro_handler().download_boletim_from_pesagro!(@storage_path, cotacao)
+          Logger.info("[#{__MODULE__}] ==> Arquivo #{file_path} baixado com sucesso da Pesagro")
+          file_path
+        end)
+      end
+
+    files_path = Task.await_many(tasks)
+
+    Logger.info("[#{__MODULE__}] ==> #{length(files_path)} arquivos foram baixados da Pesagro")
+
+    if Enum.any?(files_path, &String.ends_with?(&1, "zip")) do
+      trigger_zip_extraction()
+    else
+      trigger_pdf_upload()
+    end
+
+    {:noreply, files_path}
+  end
+
+  def handle_cast(:extract_zip, files_path) do
+    pdfs =
+      files_path
+      |> Enum.filter(&String.ends_with?(&1, "zip"))
+      |> Enum.map(fn zip_path ->
+        Task.async(fn ->
+          pesagro_handler().extract_boletins_zip!(zip_path, @storage_path)
+        end)
+      end)
+      |> Task.await_many()
+      |> List.flatten()
+
+    Logger.info("[#{__MODULE__}] ==> #{length(pdfs)} PDFs extraídos de arquivos ZIP da Pesagro")
+
+    trigger_pdf_upload()
+
+    {:noreply,
+     files_path
+     |> Enum.reject(&String.ends_with?(&1, "zip"))
+     |> Kernel.++(pdfs)}
+  end
+
+  def handle_cast(:upload_pdf, pdfs_paths) do
+    for {pdf, idx} <- Enum.with_index(pdfs_paths) do
+      Logger.info(
+        "[#{__MODULE__}] ==> Fazendo upload do PDF/Cotação #{pdf} para conversão para txt na Zamzar"
+      )
+
+      :timer.apply_after(second_to_ms(idx), zamzar_handler(), :upload_pesagro_pdf!, [pdf, idx + 2])
+    end
+
+    {:noreply, []}
+  end
+
+  def handle_cast({:download_txt, %Job{} = job}, _state) do
+    job = zamzar_api().retrieve_job!(job.id)
+
+    if ZamzarLogic.job_is_successful?(job) do
+      zamzar_handler().download_pesagro_txt!(job, @storage_path)
+
+      {:noreply, []}
+    else
+      trigger_txt_download(job, 2)
+
+      {:noreply, []}
+    end
+  end
+
+  @impl true
+  def handle_info(:schedule_convertion, state) do
+    GenServer.cast(__MODULE__, :convert)
+    {:noreply, state}
+  end
+
+  defp second_to_ms(s) do
+    s * 100 * 10
```

Comment:
Aqui eu utilizaria uma erlang function 😝 , curto demais o `:timer` para fazer conversões de tempo:

```suggestion
    :timer.seconds(s)
```